kedro/templates/project/hooks/utils.py

```diff
@@ -0,0 +1,160 @@
+from pathlib import Path
+import shutil
+import sys
+import click
+
+current_dir = Path.cwd()
+
+lint_requirements = "black~=22.12.0\nruff~=0.0.290\n"
+lint_pyproject_requirements = """
+[tool.ruff]
+select = [
+    "F",  # Pyflakes
+    "E",  # Pycodestyle
+    "W",  # Pycodestyle
+    "UP",  # pyupgrade
+    "I",  # isort
+    "PL", # Pylint
+]
+ignore = ["E501"]  # Black takes care of line-too-long
+"""
+
+test_requirements = "pytest-cov~=3.0\npytest-mock>=1.7.1, <2.0\npytest~=7.2"
+test_pyproject_requirements = """
+[tool.pytest.ini_options]
+addopts = \"\"\"
+--cov-report term-missing \\
+--cov src/{{ cookiecutter.python_package }} -ra
+\"\"\"
+
+[tool.coverage.report]
+fail_under = 0
+show_missing = true
+exclude_lines = ["pragma: no cover", "raise NotImplementedError"]
+"""
+
+docs_pyproject_requirements = """
+docs = [
+    "docutils<0.18.0",
+    "sphinx~=3.4.3",
+    "sphinx_rtd_theme==0.5.1",
+    "nbsphinx==0.8.1",
+    "nbstripout~=0.4",
+    "sphinx-autodoc-typehints==1.11.1",
+    "sphinx_copybutton==0.3.1",
+    "ipykernel>=5.3, <7.0",
+    "Jinja2<3.1.0",
+    "myst-parser~=0.17.2",
+]
+"""
+
+def _validate_range(start, end):
+    if int(start) > int(end):
+        message = f"'{start}-{end}' is an invalid range for project add-ons."
+        click.secho(message, fg="red", err=True)
+        sys.exit(1)
+
+def _validate_selection(add_ons):
+    for add_on in add_ons:
+        if int(add_on) < 1 or int(add_on) > 5:
+            message = f"'{add_on}' is not a valid selection."
+            click.secho(message, fg="red", err=True)
+            sys.exit(1)
```

Comment:
These messages don't get shown to the user:

![Screenshot 2023-09-26 at 15 41 46](https://github.com/kedro-org/kedro/assets/49397448/869480c2-1184-48a0-a75c-f54a3d9c7c30)